src/main/java/seedu/address/logic/commands/ExportCommand.java

```diff
@@ -63,25 +67,35 @@ public CommandResult execute(Model model) throws CommandException {
 
     static List<Map<String, String>> readAndParseJson(String filePath) throws IOException {
         List<Map<String, String>> jsonData = new ArrayList<>();
-        String jsonString = new String(Files.readAllBytes(Paths.get(filePath)));
-
-        Pattern personPattern = Pattern.compile("\\{([^{}]+)}");
-        Matcher personMatcher = personPattern.matcher(jsonString);
-
-        while (personMatcher.find()) {
-            String personString = personMatcher.group(1);
-            Map<String, String> personData = new LinkedHashMap<>();
-            Pattern kvPattern = Pattern.compile("\"(\\w+)\"\\s*:\\s*\"([^\"]*)\"");
-            Matcher kvMatcher = kvPattern.matcher(personString);
-            while (kvMatcher.find()) {
-                String key = kvMatcher.group(1);
-                String value = kvMatcher.group(2);
-                personData.put(key, value);
-            }
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        JsonNode rootNode = objectMapper.readTree(new File(filePath));
+        JsonNode persons = rootNode.get("persons");
+
+        for (JsonNode person : persons) {
+            Map<String, String> personInfo = new LinkedHashMap<>();
+
+            person.fields().forEachRemaining(entry -> {
+                String header = entry.getKey(); // e.g. name, email, tags, etc.
+                JsonNode value = entry.getValue(); // e.g. the value associated with the header
+
+                // isTextual checks if a JsonNode represents a basic JSON String value.
+                if (value.isTextual()) {
+                    personInfo.put(header, value.asText());
+                // Note that the tags in AddressBook.json are stored in an array literal.
```

Comment:
Nice find! It's good to account for the various possible data structures when parsing in our case. Perhaps we can have some internal documentation for this in the future (i.e. String literals, Array of Objects, Map etc)